chainstate/tx-verifier/src/transaction_verifier/signature_check.rs

```diff
@@ -53,18 +53,22 @@ where
         .collect::<Result<Vec<_>, ConnectTransactionError>>()?;
     let inputs_utxos = inputs_utxos.iter().map(|utxo| utxo.as_ref()).collect::<Vec<_>>();
 
-    inputs.iter().enumerate().try_for_each(|(input_idx, input)| {
-        // TODO: ensure that signature verification is tested in the test-suite, they seem to be tested only internally
-        let destination = destination_getter.call(input)?;
-        verify_signature(
-            chain_config,
-            &destination,
-            transactable,
-            &inputs_utxos,
-            input_idx,
-        )
-        .map_err(ConnectTransactionError::SignatureVerificationFailed)
-    })
+    inputs
+        .iter()
+        .zip(transactable.signatures().unwrap().iter())
```

Comment:
It's an invariant that witness size is equal to input size. You can use `zip_eq` from itertools. But before using it, make an assertion before that asserts it.